### DISCUSSION TITLE
Trick terraform manager into scanning file

### DISCUSSION THREAD

**Discussion by ekhaydarov** [renovatebot/renovate#11860] [Request Help] [answered]
I have a terragrunt `versions-extra.hcl` that looks this. it lets terragrunt update terraforms `version.tf` with that extra package.
```
locals {
  additional_provider_versions = {
    null = {
      source  = "hashicorp/null"
      version = "3.0.0"
    }
  }
}
```
I have added that file to be included in the fileMatch as so. DEBUG logs add the file to the log `Matched 1225 file(s) for manager terraform:` but then there is nothing for that file after. I assumed it got rejected because it was not found in [contentCheckList](https://github.com/renovatebot/renovate/blob/main/lib/manager/terraform/extract.ts#L30) so I added the `required_provider` key to the locals in the `versions-extra.hcl` to try and find the file again. It still does not recognise the file.
```
  terraform: {
    fileMatch: ["(^|/)versions\\.tf$", "(^|/)terragrunt\\.hcl$", "(^|/)versions-extra\\.hcl$"],
    ignorePaths: ["**/main.tf"]
  },
```

Would this file be parseable by terraform manager at all? If not what are my options other than rewriting terragrunt?
Thanks

**Comment by rarkins** [comment, marked as answer]
You could use the [regex manager](https://docs.renovatebot.com/configuration-options/#regexmanagers) for any custom extract purposes

**Reply by ekhaydarov** [reply]
Thanks for the suggestion, I am just looking into this and need some guidance. Based on what terraform manager spits out currently
```
locals {
  additional_provider_versions = {
    databricks = {
      source  = "databrickslabs/databricks"
      version = "=0.3.1"
    }
    random = {
      source  = "hashicorp/random"
      version = "=3.0.0"
    }
    null = {
      source  = "hashicorp/null"
      version = "3.0.0"
    }
  }
}
```
```
{
                 "currentValue": "=2.59.0",
                 "depType": "required_provider",
                 "depName": "azurerm",
                 "datasource": "terraform-provider",
                 "lookupName": "hashicorp/azurerm",
                 "depIndex": 1,
                 "updates": [
                   {
                     "bucket": "non-major",
                     "newVersion": "2.70.0",
                     "newValue": "=2.70.0",
                     "releaseTimestamp": "2021-07-30T01:37:57.000Z",
                     "newMajor": 2,
                     "newMinor": 70,
                     "updateType": "minor",
                     "isRange": true,
                     "branchName": "renovate/azurerm-2.x"
                   }
                 ],
                 "warnings": [],
                 "versioning": "hashicorp",
                 "sourceUrl": "https://github.com/terraform-providers/terraform-provider-azurerm",
                 "homepage": "https://registry.terraform.io/providers/hashicorp/azurerm",
                 "currentVersion": "2.59.0",
                 "isSingleVersion": true,
                 "fixedVersion": "2.59.0"
               }
``` 
I am trying to find the depName after `/`. testing on regex101 for js seems to work. However logs state that no files matched, even though its looking through the right files. I am not sure what im missing with this config?
```
regexManagers: [
    {
      fileMatch: ["(^|/)versions-extra\\.hcl$"],
      matchStrings: ["\"source\":\\s*\"(.*?)\\/(?<depName>.*)\"[^\"]*\"version\":\\s*\"(?<currentValue>.*)\""],
      versioningTemplate: "hashicorp",
      datasourceTemplate: "terraform-provider"
    }
  ],
```

**Reply by rarkins** [reply]
@secustor do you have any advice here, e.g. whether multiple `matchStrings` would simplify it or a single match is best?

**Reply by ekhaydarov** [reply]
actually just realised I'm matching for semicolons like the examples in readme instead of the .hcl file which has `=`. also no double quotes. it was pretty late last night. fresh pair of eyes in the morning work better

**Reply by ekhaydarov** [reply]
works great. manager looks like this for any future searchers
```
  regexManagers: [
    {
      fileMatch: ["(^|/)versions-extra\\.hcl$"],
      matchStrings: [
        "source\s*=\s*"(?<lookupName>.*\/(?<depName>.*))"[^"]*version\s*=\s*"(?<currentValue>.*)""
      ],
      versioningTemplate: "hashicorp",
      datasourceTemplate: "terraform-provider"
    }
  ]
```

Edit: actually need to extract `lookupName`, so need to change the regex to pick it up

**Reply by rarkins** [reply]
Thank you for sharing your answer!
